### PR TITLE
fix(DeleteLinks):provided an option to delete links

### PR DESCRIPTION
**Pull request by ponrajaprabhusk**
### Functionality:
provided an option to delete links

### Solution:
provided an option to delete the links and unlink the PR in task details.

### Risk level:
- [ ] high 
- [X] medium
- [ ] low

### How to test:
by adding another PR or a link and deleting them

### CODE REVIEW

**Review by viv9k — COMMENTED** [review]
Remove link mechanism is not needed just remove link doc is enough

File: functions/model/linker/lib.js [hunk 1]
```diff
@@ -48,6 +48,39 @@ exports.setLinkDoc = function(orgDomain, taskId, linkType, linkURL, linkID) {
...
+ * @param {any} orgDomain
+ * @param {any} taskId
+ * @param {any} linkID
+ * @return {any}
+ */
+
+exports.getLinkData = function(orgDomain, taskId, linkId){
+    console.log(orgDomain)
+    console.log(taskId)
+    console.log(linkId)
```
Comment (viv9k) [inline functions/model/linker/lib.js:83]: Remove console logs

File: functions/model/linker/linker.js [hunk 2]
```diff
@@ -46,6 +47,17 @@ fastify.post("/setLink", (req, res) => {
   setLinkDetails(req, res);
 });
 
+/**
+ * Description
+ * @param {any} "/removeLink"
+ * @param {any} (req
```
Comment (viv9k) [inline functions/model/linker/linker.js:53]: req should not have starting bracket

File: functions/model/linker/tark/removeLink.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+ const { getTask, updateTask } = require("../../tasks/lib");
+ const { removeLinkDoc, getLinkData } = require("../lib")
+ 
+ exports.removeLink = function(request, response) {
+     const orgDomain = request.body.data.OrgDomain;
+     const taskId = request.body.data.TaskId;
+     const linkId = request.body.data.LinkId;
+     const linkType = request.body.data.LinkType;
+
+    console.log(orgDomain, taskId, linkId , linkType)
```
Comment (viv9k) [inline functions/model/linker/tark/removeLink.js:32]: remove console.log

File: functions/model/linker/tark/removeLink.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+ 
+     let status = 200;
+     let result;
+     const promise = getTask(taskId, orgDomain).then((taskDetail) => {
+         if (taskDetail == undefined) {
+             result = {data: {status: 500}};
+         } else {
+            console.log('this is hitting')
```
Comment (viv9k) [inline functions/model/linker/tark/removeLink.js:40]: remove console.log

File: src/app/body/task-details/task-details.component.ts [hunk 5]
```diff
@@ -212,6 +212,7 @@ export class TaskDetailsComponent implements OnInit {
         return actions.data as Link[];
     })).subscribe({
       next: (data) => {
+        console.log("Data Link", data);
```
Comment (viv9k) [inline src/app/body/task-details/task-details.component.ts:215]: remove console.log

File: src/app/body/task-details/task-details.component.ts [hunk 6]
```diff
@@ -221,6 +222,27 @@ export class TaskDetailsComponent implements OnInit {
     });
   }
 
+  removeLink(linkId, linkType){
+    console.log(linkId, linkType, this.orgDomain, this.Id)
```
Comment (viv9k) [inline src/app/body/task-details/task-details.component.ts:226]: remove console.log

File: src/app/body/task-details/task-details.component.ts [hunk 7]
```diff
@@ -221,6 +222,27 @@ export class TaskDetailsComponent implements OnInit {
...
+    const callable = this.functions.httpsCallable('linker/removeLink');
+    callable({ OrgDomain: this.orgDomain, TaskId: this.Id, LinkType: linkType, LinkId:linkId  }).subscribe({
+      next: (data) => {
+        return;
+      },
+      error: (error) => {
+        this.errorHandlerService.showError = true;
+        this.errorHandlerService.getErrorCode(this.componentName, "InternalError", "Api");
+        console.log("Error", error);
+        console.error(error);
```
Comment (viv9k) [inline src/app/body/task-details/task-details.component.ts:236]: They are same

**Review by viv9k — APPROVED** [review]
Please fix the code, remove unnecessary elements

**Comment by viv9k** [comment]
Too much of data leak is happening, please fix them

File: functions/model/linker/lib.js (hunk 1 above)
Comment (ponrajaprabhusk) [reply functions/model/linker/lib.js:83]: Done

**Comment by ponrajaprabhusk** [comment]
> Too much of data leak is happening, please fix them

Fixed !!

**Comment by viv9k** [comment]
Approved

**Pull request merged**